x-pack/plugins/security_solution/public/users/pages/users_tabs.tsx

```diff
@@ -5,40 +5,78 @@
  * 2.0.
  */
 
-import React, { memo } from 'react';
+import React, { memo, useCallback } from 'react';
 import { Route, Switch } from 'react-router-dom';
 
 import { UsersTabsProps } from './types';
 import { UsersTableType } from '../store/model';
 import { USERS_PATH } from '../../../common/constants';
 import { AllUsersQueryTabBody } from './navigation';
+import { AnomaliesQueryTabBody } from '../../common/containers/anomalies/anomalies_query_tab_body';
+import { AnomaliesUserTable } from '../../common/components/ml/tables/anomalies_user_table';
+import { Anomaly } from '../../common/components/ml/types';
+import { scoreIntervalToDateTime } from '../../common/components/ml/score/score_interval_to_datetime';
+import { UpdateDateRange } from '../../common/components/charts/common';
 
 export const UsersTabs = memo<UsersTabsProps>(
   ({
     deleteQuery,
-    docValueFields,
     filterQuery,
     from,
     indexNames,
     isInitializing,
     setQuery,
     to,
     type,
+    setAbsoluteRangeDatePicker,
   }) => {
+    const narrowDateRange = useCallback(
+      (score: Anomaly, interval: string) => {
+        const fromTo = scoreIntervalToDateTime(score, interval);
+        setAbsoluteRangeDatePicker({
+          id: 'global',
+          from: fromTo.from,
+          to: fromTo.to,
+        });
+      },
+      [setAbsoluteRangeDatePicker]
+    );
+
+    const updateDateRange = useCallback<UpdateDateRange>(
+      ({ x }) => {
+        if (!x) {
+          return;
+        }
+        const [min, max] = x;
+        setAbsoluteRangeDatePicker({
+          id: 'global',
+          from: new Date(min).toISOString(),
+          to: new Date(max).toISOString(),
+        });
+      },
+      [setAbsoluteRangeDatePicker]
+    );
+
+    const tabProps = {
+      deleteQuery,
+      endDate: to,
+      filterQuery,
+      indexNames,
+      skip: isInitializing || filterQuery === undefined,
+      setQuery,
+      startDate: from,
+      type,
+      narrowDateRange,
+      updateDateRange,
+    };
+
     return (
       <Switch>
         <Route path={`${USERS_PATH}/:tabName(${UsersTableType.allUsers})`}>
-          <AllUsersQueryTabBody
-            deleteQuery={deleteQuery}
-            endDate={to}
-            filterQuery={filterQuery}
-            indexNames={indexNames}
-            skip={isInitializing || filterQuery === undefined}
-            setQuery={setQuery}
-            startDate={from}
-            type={type}
-            docValueFields={docValueFields}
-          />
+          <AllUsersQueryTabBody {...tabProps} />
+        </Route>
+        <Route path={`${USERS_PATH}/:tabName(${UsersTableType.anomalies})`}>
+          <AnomaliesQueryTabBody {...tabProps} AnomaliesTableComponent={AnomaliesUserTable} />
```

Comment:
Could we change the AnomaliesQueryTabBody component (`security_solution/public/common/containers/anomalies/anomalies_query_tab_body/index.tsx`) to determine which component to use for `AnomaliesTableComponent` by the `tabProps.type`? Then we don't have to do `// eslint-disable-next-line` that's on that component prop and we reduce the amount of props we're passing.

Reply:
I am not sure if I understand your proposal. Are you proposing to instead of providing `AnomaliesTableComponent` as a prop to create a function inside `AnomaliesQueryTabBodyComponent` with a switch-case? Like:
```js  switch (key) {
switch (key) {
    case UsersType:
      return AnomaliesUserTable;
    case NetworkType:
      return AnomaliesNetworkTable;
    case HostsType:
      return AnomaliesHostTable;
  }
  ```

If that is the case, how can I differentiate `UsersType`, `NetworkType` from `HostsType`? They have exactly the same content.

Reply:
Yes `AnomaliesQueryTabBodyComponent` has a prop of `type` passed to it that can be used to identify the type

Reply:
The value inside `type` is  `'page' | 'details'`